tools/github-actions/audit/src/reports.ts

```diff
@@ -0,0 +1,179 @@
+  warnings: OtterAdvisory[];
+  highestSeverityFound?: Severity;
+}
+
+/**
+ * Method which formats the response of an audit api (yarn or npm) into a report that can be displayed in the
+ * GitHub action summary
+ */
+export type OtterAuditReporter = (response: string, severityThreshold: Severity) => OtterAuditReport;
+
+function computeTotalsForSeverities(severityAccessor: (severity: Severity) => number) {
+  let highestSeverityFound: Severity | undefined;
+  let nbVulnerabilities = 0;
+  for (let index = severities.length - 1; index >= 0; index--) {
+    const severity: Severity = severities[index];
+    const severityTotal = severityAccessor(severity);
+    if (!highestSeverityFound && severityTotal > 0) {
+      highestSeverityFound = severity;
+    }
+    nbVulnerabilities += severityTotal;
+  }
+  return { highestSeverityFound, nbVulnerabilities };
+}
+
+function updateReportWithVulnerability(currentReport: OtterAuditReport, severityThreshold: Severity, vulnerability: OtterAdvisory): OtterAuditReport {
+  const severityThresholdIndex = severities.indexOf(severityThreshold);
+  const severityIndex = severities.indexOf(vulnerability.severity);
+  if (severityIndex >= severityThresholdIndex) {
+    currentReport.errors.push(vulnerability);
+  } else {
+    currentReport.warnings.push(vulnerability);
+  }
+  return currentReport;
+}
+
+/**
+ * Format the response from npm audit V1 in a common interface that will be used to build the
+ * report summary
+ * @param response
+ * @param severityThreshold
+ */
+export const computeNpmV1Report: OtterAuditReporter = (response: string, severityThreshold: Severity) => {
+  const reportJson = JSON.parse(response) as Audit;
+  core.debug(response);
+  const { nbVulnerabilities, highestSeverityFound } = computeTotalsForSeverities((severity) => reportJson.metadata.vulnerabilities[severity]);
+  return Object.values(reportJson.advisories as Readonly<Record<GitHubAdvisoryId, Advisory>>)
+    .reduce<OtterAuditReport>((currentVulnerabilities, advisory: Advisory) =>
+      updateReportWithVulnerability(currentVulnerabilities, severityThreshold, {
+        severity: advisory.severity,
+        overview: advisory.overview,
+        moduleName: advisory.module_name
+      })
+    , {errors: [], warnings: [], nbVulnerabilities, highestSeverityFound} as OtterAuditReport);
+};
+
+/**
+ * Format the response from npm audit V2 in a common interface that will be used to build the
+ * report summary
+ * @param response
+ * @param severityThreshold
+ */
+export const computeNpmV2Report: OtterAuditReporter = (response: string, severityThreshold: Severity) => {
+  const reportJson = JSON.parse(response) as NPMAuditReportV2.Audit;
+  core.debug(response);
+  const { nbVulnerabilities, highestSeverityFound } = computeTotalsForSeverities((severity) => reportJson.metadata.vulnerabilities[severity]);
+  return Object.values(reportJson.vulnerabilities)
+    .reduce<OtterAuditReport>((currentVulnerabilities, advisory) =>
```

Comment:
I think the type is not needed (same in other similar places)
```suggestion
    .reduce((currentVulnerabilities, advisory) =>
```

Reply:
it seems to be needed, I've removed the cast on the acc though

Reply:
This is because of the removal of the `as OtterAuditReport` in the initial value parameter, which is fine as well